Archive/Imo/Imo2021Q1.lean

```diff
@@ -56,9 +57,8 @@ lemma exists_numbers_in_interval {n : ℕ} (hn : 100 ≤ n) :
   rw [← Nat.sub_add_cancel hn'] at h₁ h₂ h₃
   set l := (n + 1).sqrt - 1
   refine ⟨l, ?_, ?_⟩
-  · calc n + 4 * l ≤ (l ^ 2 + 4 * l + 2) + 4 * l := by linarith only [h₂]
-      _ ≤ 2 * l ^ 2 := by nlinarith only [h₃]
-  · linarith only [h₁]
+  · linear_combination h₂ + (l + 1) * h₃
```

Comment:
Particularly the first looks good to me; the second feels fine.

Reply:
Moved the first one to #19352 (with a collection of other `nlinarith` replacements, to benchmark that specifically).